posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+
+
+[source, XML]
+----
+    <ejbContainer bindToServerRoot="false"/>
+----
+
+It is also possible to disable just the legacy short form default JNDI name support (i.e. the bean is bound using the interface name) with the following setting in server.xml:
+
+
+[source, XML]
+----
+   <ejbContainer disableShortDefaultBindings="true"/>
+----
+
+Since the new support for customer JNDI names and legacy defaults provider alternative JNDI names, it is now possible to disable the EJB specification required JNDI names. This is done in server.xml as follows:
+
+
+[source, XML]
+----
+   <ejbContainer bindToJavaGlobal="false"/>
+----
+
+Finally, the following new configuration attribute on the `<ejbContainer>` element in open-liberty enables the WebSphere traditional behavior of failing application start when multiple beans are bound to the same JNDI name:
+
+[source, XML]
+----
+    <ejbContainer customBindingsOnError="FAIL"/>
+----
+
+
+[#grpc]
+== Provide and consume gRPC services from your web applications
+
+link:https://grpc.io/docs/what-is-grpc/introduction/[gRPC] is a high-performance, open source universal RPC framework. gRPC support on Liberty allows developers to both provide and consume gRPC services from your web applications. The introduction of gRPC support in Open Liberty now means you can take advantage of the benefits of gRPC more easily than before. Those benefits include great performance, simple service definitions via Protocol Buffers, cross-platform and language support, and wide industry adoption.
+
+These two features which were previously in our beta are now available in this months GA: `grpc-1.0`, which enables gRPC services, and `grpcClient-1.0`, which enables the use of a gRPC client for outbound calls.
+
+The `grpc-1.0` feature works by scanning web apps for gRPC service implementations, through implementors of `io.grpc.BindableService`. The web app must include the protocol buffer compiler-generated code for the services it intends to provide, and additionally the service class must provide a no-argument constructor. The web app does not need to include any core gRPC libraries; those are provided by the Liberty runtime. Once a gRPC service is scanned and started, it becomes accessible to remote gRPC clients on the configured HTTP ports.
+
+The `grpcClient-1.0` feature provides applications with access to a link:https://netty.io/[Netty] gRPC client, as well as the related libraries. A web app must provide a client implementation and stubs, and can make outbound calls with a `io.grpc.ManagedChannel` without needing to provide the supporting client libraries.
+
+You can now sample gRPC by either following the instructions in this link:https://github.com/OpenLiberty/sample-grpc[GitHub repositories] README or following this basic Hello World service (add the  `grpc-1.0` to the `server.xml`):
```

Comment:
`You can now sample gRPC` doesn't make any sense.  Maybe it should be `You can now try gRPC`.  `GitHub repositories` seems like it should be `GitHub repository's`.  `add the grpc-1.0 to the server.xml` should be `add the grpc-1.0 feature to your server.xml`

Reply:
Done